sdk/identity/Azure.Identity/tests/samples/UserAuthenticationSnippets.cs

```diff
@@ -120,41 +120,50 @@ public static async Task<TokenCredential> GetUserCredentialAsync()
+                // We can also optionally specify a name for the cache to avoid having it cleared by other applications.
                 credential = new InteractiveBrowserCredential(
-                    new InteractiveBrowserCredentialOptions
-                    {
-                        TokenCachePersistenceOptions = new TokenCachePersistenceOptions()
-                    });
+                    new InteractiveBrowserCredentialOptions { TokenCachePersistenceOptions = new TokenCachePersistenceOptions { Name = TOKEN_CACHE_NAME } });
 
-                AuthenticationRecord authRecord = await credential.AuthenticateAsync();
+                // Call AuthenticateAsync to fetch a new AuthenticationRecord.
+                authRecord = await credential.AuthenticateAsync();
 
-                using (var authRecordStream = new FileStream(AUTH_RECORD_PATH, FileMode.Create, FileAccess.Write))
-                {
-                    await authRecord.SerializeAsync(authRecordStream);
-                }
+                // Serialize the AuthenticationRecord to disk so that it can be re-used across executions of this initialization code.
+                using var authRecordStream = new FileStream(AUTH_RECORD_PATH, FileMode.Create, FileAccess.Write);
+                await authRecord.SerializeAsync(authRecordStream);
             }
             else
             {
-                AuthenticationRecord authRecord;
+                // Load the previously serialized AuthenticationRecord from disk and deserialize it.
+                using var authRecordStream = new FileStream(AUTH_RECORD_PATH, FileMode.Open, FileAccess.Read);
+                authRecord = await AuthenticationRecord.DeserializeAsync(authRecordStream);
+            }
 
-                using (var authRecordStream = new FileStream(AUTH_RECORD_PATH, FileMode.Open, FileAccess.Read))
+            // Construct a new client with our TokenCachePersistenceOptions with the addition of the AuthenticationRecord property.
+            // This tells the credential to use the same token cache in addition to which account to try and fetch from cache when GetToken is called.
+            credential = new InteractiveBrowserCredential(
+                new InteractiveBrowserCredentialOptions
                 {
-                    authRecord = await AuthenticationRecord.DeserializeAsync(authRecordStream);
-                }
-
-                credential = new InteractiveBrowserCredential(
-                    new InteractiveBrowserCredentialOptions
-                    {
-                        TokenCachePersistenceOptions = new TokenCachePersistenceOptions(),
-                        AuthenticationRecord = authRecord
-                    });
-            }
+                    TokenCachePersistenceOptions = new TokenCachePersistenceOptions { Name = TOKEN_CACHE_NAME },
+                    AuthenticationRecord = authRecord
+                });
```

Comment:
why not move this into the else statement above so you're not creating 2 credential instances in the case the auth record doesn't exist on disk?

Reply:
My assumption here is that the first client has been constructed with an options instance that does not include the AuthenticationRecord.  The second client creates a new options instance that populates the AuthenticationRecord property. But perhaps this is not necessary since we already called Authenticate which populates the cache and sets the AR internally?